### DISCUSSION TITLE
Install new php modules using pecl

### DISCUSSION THREAD

**Discussion by sdoherty0** [FaserF/hassio-addons#101] [General]
Hi, is there a way to get access to this container to install pecl (or if it is already installed) to install new php modules? i need to install ssh2.
Thanks,
Scott

**Comment by FaserF** [comment]
I have prepared support with the latest commit. But it is not working yet for me.
https://github.com/FaserF/hassio-addons/blob/master/apache2/Dockerfile#L21
You can change the dockerfile always for yourself for your needs. But there is no nice way to enter the container from within HA for yourself.

**Reply by FaserF** [reply]
I have seen that you have tried to implement it by yourself right now. If you are having a working release with PHP8.1 I am open for a pull request from your side :)

**Reply by ristoksi** [reply]
is this feature now working? I should install PHPMailer, how to do that?

**Reply by ristoksi** [reply]
After chatting with ChatGPT I managed to get it working with following init_commands

init_commands:
  - wget -O /tmp/composer-setup.php https://getcomposer.org/installer
  - php /tmp/composer-setup.php --install-dir=/usr/local/bin --filename=composer
  - composer require phpmailer/phpmailer --working-dir=/share/htdocs

I suppose I can now remove these init_commands. At least it seems to work.

**Comment by benjaminmiller99** [comment]
> ##
You could try asking an AI — it might be able to solve your problem.

**Comment by FaserF** [comment]
This feature is already supported and resolved in the Apache2 add-ons!

You have two main ways to use custom PHP extensions, Composer packages, or PECL modules:

1. Pre-installed PHP Extensions
The add-on comes with a comprehensive list of standard PHP modules pre-installed out of the box (e.g. curl, gd, intl, mbstring, mysqli, pdo_mysql, zip, soap, pecl-redis, pecl-imagick, pecl-imap, pecl-apcu, sockets, sodium, etc.).

2. Using init_commands for Additional Modules or Composer
You can use the init_commands array in the add-on configuration tab to run custom startup commands when the container initializes.

To install additional Alpine PHP packages (e.g., ssh2):


init_commands:
  - apk add --no-cache php85-pecl-ssh2
To install Composer & packages like phpmailer automatically:

init_commands:
  - wget -O /tmp/composer-setup.php https://getcomposer.org/installer
  - php /tmp/composer-setup.php --install-dir=/usr/local/bin --filename=composer
  - composer require phpmailer/phpmailer --working-dir=/share/htdocs
Once added to init_commands, these will persist across add-on restarts.
